lib/storage/providers/MemoryOnlyProvider.js

```diff
@@ -0,0 +1,155 @@
+import _ from 'underscore';
+import sizeof from 'object-sizeof';
+import utils from '../../utils';
+
+// eslint-disable-next-line import/no-mutable-exports
+let store = {};
+
+const setInternal = (key, value) => {
+    store[key] = value;
+    return Promise.resolve(value);
+};
+
+const isJestRunning = typeof jest !== 'undefined';
+const set = isJestRunning ? jest.fn(setInternal) : setInternal;
+
+const provider = {
+    /**
+     * The name of the provider that can be printed to the logs
+     */
+    name: 'MemoryOnlyProvider',
+
+    /**
+     * Initializes the storage provider
+     */
+    init() {},
+
+    /**
+     * Get the value of a given key or return `null` if it's not available in memory
+     * @param {String} key
+     * @return {Promise<*>}
+     */
+    getItem(key) {
+        return Promise.resolve(store[key]);
+    },
+
+    /**
+     * Get multiple key-value pairs for the give array of keys in a batch.
+     * @param {String[]} keys
+     * @return {Promise<Array<[key, value]>>}
+     */
+    multiGet(keys) {
+        const getPromises = _.map(keys, (key) => new Promise((resolve) => this.getItem(key).then((value) => resolve([key, value]))));
+        return Promise.all(getPromises);
+    },
+
+    /**
+     * Sets the value for a given key. The only requirement is that the value should be serializable to JSON string
+     * @param {String} key
+     * @param {*} value
+     * @return {Promise<void>}
+     */
+    setItem(key, value) {
+        return set(key, value);
+    },
+
+    /**
+     * Stores multiple key-value pairs in a batch
+     * @param {Array<[key, value]>} pairs
+     * @return {Promise<void>}
+     */
+    multiSet(pairs) {
+        const setPromises = _.map(pairs, ([key, value]) => this.setItem(key, value));
+        return new Promise((resolve) => Promise.all(setPromises).then(() => resolve(store)));
```

Comment:
Does this actually need to resolve with the `store` as the result? The jsDoc labels this as a `void` return

Reply:
yes indeed